synapse/static/client/register/js/register.js

```diff
@@ -1,67 +1,93 @@
 window.matrixRegistration = {
-    endpoint: location.origin + "/_matrix/client/api/v1/register"
+    endpoint: location.origin + "/_matrix/client/r0/register"
 };
 
 var setupCaptcha = function() {
-    if (!window.matrixRegistrationConfig) {
-        return;
-    }
-    $.get(matrixRegistration.endpoint, function(response) {
-        var serverExpectsCaptcha = false;
-        for (var i=0; i<response.flows.length; i++) {
-            var flow = response.flows[i];
-            if ("m.login.recaptcha" === flow.type) {
-                serverExpectsCaptcha = true;
-                break;
+    $.ajax({
+        url: matrixRegistration.endpoint,
+        method: "POST",
+        data: '{}',
+        complete: function(response) {
+            if (response.status != 401) {
+                errorFunc(response);
+                return;
             }
+            response = response.responseJSON;
+            window.matrixRegistration.session = response.session;
+            var serverExpectsCaptcha = false;
+            for (var i=0; i<response.flows.length; i++) {
+                var flow = response.flows[i];
+                for (j=0; j<flow.stages.length; j++) {
+                    var stage = flow.stages[j];
+                    if ("m.login.recaptcha" === stage) {
+                        serverExpectsCaptcha = true;
+                        break;
+                    }
+                }
+            }
+            if (!serverExpectsCaptcha) {
+                console.log("This server does not require a captcha.");
+                return;
```

Comment:
... we should still complete the other auth stages, and call the onRegistered callback though?